app/controllers/orders_controller.rb

```diff
@@ -20,21 +20,25 @@ def show
     city = @customer.city || 'Chicago'
     postal_code = @customer.zip_code || '60609'
 
-
-    @order = Stripe::Order.create(
-      :currency => 'usd',
-      :customer => @customer.stripe_customer_id,
-      :items => items,
-      :shipping => {
-        :name => first_name,
-        :address => {
-          :line1 => line1,
-          :city => city,
-          :country => 'US',
-          :postal_code => postal_code
-        }
-      },
-    )
+    begin
+      @order = Stripe::Order.create(
+        :currency => 'usd',
+        :customer => @customer.stripe_customer_id,
+        :items => items,
+        :shipping => {
+          :name => first_name,
+          :address => {
+            :line1 => line1,
+            :city => city,
+            :country => 'US',
+            :postal_code => postal_code
+          }
+        },
+      )
+    rescue => error
+      flash[:warning] = "Invalid Shipping Address"
+      redirect_to ('/')
```

Comment:
This `redirect_to` should just be temporary. Welll need to find a better way to handle the shipping error if a customer has an invalid address.